docs/troubleshooting.md

```diff
@@ -0,0 +1,38 @@
+# Troubleshooting Tips
+
+When developing operators, there might occasionally be moments when Kubernetes
+APIs or apiserver don't return error message with full context or when the error
+message can be a bit misleading. Purpose of this document is to list some common
+pitfalls that might be hard to debug in some cases and where returned error
+message is not perfectly clear.
+
+## RBAC rules
+
+When `informer.Watch()` is called, it might fail with an "unknown error". This
+error is not very descriptive and there is work ongoing towards OperatorKit
+improving logging on this as the original error is
+[StatusError](https://github.com/kubernetes/apimachinery/blob/master/pkg/api/errors/errors.go#L39)
+with `ErrStatus.Reason`:
+[Forbidden](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L569)
```

Comment:
Not sure about linking master instead of a permalink. It could easily move to the wrong line here.